src/prebid.js

```diff
@@ -642,11 +643,13 @@ $$PREBID_GLOBAL$$.requestBids = (function() {
     req.metrics.checkpoint('requestBids');
     let adUnits = req.adUnits || $$PREBID_GLOBAL$$.adUnits;
     req.adUnits = (isArray(adUnits) ? adUnits.slice() : [adUnits]);
-    return delegate.call(this, req);
+    req.defer = defer({promiseFactory: (r) => new Promise(r)})
+    delegate.call(this, req);
+    return req.defer.promise;
```

Comment:
Maybe a few comments why this is necessary would be helpful.

As someone only looking at bid adapters mostly the fun-hook framework and the custom promise implementation this code is non-obivous why it's there.

Reply:
If you're asking about the reason for fun-hooks or the custom promise:

 - fun-hooks is basically a plugin system (to the author: sorry if I'm oversimplifying this). if we have 5 modules that may be loaded in a random combination and order, and all plug into the same code path, fun-hook solves the problem of orchestrating those. I can't answer why we are using that over something else.
 - the custom promise - `A version of Promise that runs callbacks synchronously when it can` - came about because of [this discussion](https://github.com/prebid/Prebid.js/pull/8106#pullrequestreview-1003052163).

As for this code specifically, I tried clarifying it in the comments - it's boils down to idiosyncrasies of fun-hooks.

Reply:
Thanks for the clarification :hugs: